shell/edgexfoundry-snapcraft.sh

```diff
@@ -0,0 +1,175 @@
+#!/bin/bash -x
+
+# note that since this script gets included into a yaml string using jjb's
+# include-raw-escape macro, we can't use any squiggly braces here, otherwise
+# they will get duplicated
+# for some reason we also can't use include-raw macro because that macro will
+# attempt to interpret anything contianed in a macro as a jenkins key to
+# expand which we also don't want
+
+# if we are running inside a jenkins instance then copy the login file 
+# and check if this is a release job
+if [ -n "$JENKINS_URL" ]; then
+    # since we are running from jenkins, use the workspace as the git root
+    GIT_ROOT=$WORKSPACE
+    if [ -f "$HOME/EdgeX" ]; then
+       cp "$HOME/EdgeX" "$GIT_ROOT/edgex-snap-store-login"
+    else
+        echo "I seem to be running on Jenkins, but there's not a snap store login file..." 
+    fi
+
+    # figure out what kind of job this is using $JOB_NAME and simplify that 
+    # into $JOB_TYPE
+    if [[ "$JOB_NAME" =~ .*-snap-.*-stage-snap.* ]]; then
+        JOB_TYPE="stage"
+    elif [[ "$JOB_NAME" =~ .*-snap-release-snap ]]; then
+        JOB_TYPE="release"
+    else
+        JOB_TYPE="build"
+    fi
+else
+    # not running in a jenkins container, assume the current working directory
+    # is the repo git dir that we want
+    GIT_ROOT=$PWD
+fi
+
+# find the name of this snap from the snapcraft.yaml "name" key
+# note that we assume that snapcraf.yaml is located in snap/snapcraft.yaml 
```

Comment:
*snapcraft.yaml
Also, this is a minor quibble, but the combination of no caps at the beginning of sentences and no periods at the end of some sentences (where EOL seems to be an implicit end of sentence) makes this a bit confusing to read. Readability could certainly be improved.

Reply:
Fixed (and changed all of the other comments in this file)